statsmodels/stats/rates.py

```diff
@@ -341,19 +1108,660 @@ def tost_poisson_2indep(count1, exposure1, count2, exposure2, low, upp,
+    """
+    # TODO: avoid possible circular import, check if needed
+    from statsmodels.stats.power import normal_power_het
+
+    nobs_ratio = nobs2 / nobs1
+    v1 = dispersion / exposure * (1 / rate1 + 1 / (nobs_ratio * rate2))
+    if method_var == "alt":
+        v0 = v1
+    elif method_var == "score":
+        # nobs_ratio = 1 / nobs_ratio
+        v0 = dispersion / exposure * (1 + value / nobs_ratio)**2
+        v0 /= value / nobs_ratio * (rate1 + (nobs_ratio * rate2))
+    else:
+        raise NotImplementedError(f"method_var {method_var} not recognized")
+
+    std_null = np.sqrt(v0)
+    std_alt = np.sqrt(v1)
+    es = np.log(rate1 / rate2) - np.log(value)
+
+    pow_ = normal_power_het(es, nobs1, alpha, std_null=std_null,
+                            std_alternative=std_alt,
+                            alternative=alternative)
+
+    p_pooled = None  # TODO: replace or remove
+
+    if return_results:
+        res = HolderTuple(
+            power=pow_,
+            p_pooled=p_pooled,
+            std_null=std_null,
+            std_alt=std_alt,
+            nobs1=nobs1,
+            nobs2=nobs_ratio * nobs1,
+            nobs_ratio=nobs_ratio,
+            alpha=alpha,
+            tuple_=("power",),  # override default
+            )
+        return res
+
+    return pow_
+
+
+def power_equivalence_poisson_2indep(rate1, nobs1, rate2, nobs2, exposure,
+                                     low, upp, alpha=0.05, dispersion=1,
+                                     method_var="alt"):
+    """power for equivalence test of ratio of 2 independent poisson rates
+
+    References
+    ----------
+    .. [1] Zhu, Haiyuan. 2017. “Sample Size Calculation for Comparing Two
+       Poisson or Negative Binomial Rates in Noninferiority or Equivalence
+       Trials.” Statistics in Biopharmaceutical Research, March.
+       https://doi.org/10.1080/19466315.2016.1225594
+    .. [2] Zhu, Haiyuan, and Hassan Lakkis. 2014. “Sample Size Calculation for
+       Comparing Two Negative Binomial Rates.” Statistics in Medicine 33 (3):
+       376–87. https://doi.org/10.1002/sim.5947.
+    .. [3] PASS documentation
+    """
+    nobs_ratio = nobs2 / nobs1
+    v1 = dispersion / exposure * (1 / rate1 + 1 / (nobs_ratio * rate2))
+
+    if method_var == "alt":
+        v0_low = v0_upp = v1
+    elif method_var == "score":
+        v0_low = dispersion / exposure * (1 + low * nobs_ratio)**2
+        v0_low /= low * nobs_ratio * (rate1 + (nobs_ratio * rate2))
+        v0_upp = dispersion / exposure * (1 + upp * nobs_ratio)**2
+        v0_upp /= upp * nobs_ratio * (rate1 + (nobs_ratio * rate2))
+    else:
+        raise NotImplementedError(f"method_var {method_var} not recognized")
+
+    es_low = np.log(rate1 / rate2) - np.log(low)
+    es_upp = np.log(rate1 / rate2) - np.log(upp)
+    std_null_low = np.sqrt(v0_low)
+    std_null_upp = np.sqrt(v0_upp)
+    std_alternative = np.sqrt(v1)
+
+    pow_ = _power_equivalence_het(es_low, es_upp, nobs2, alpha=alpha,
```

Comment:
nobs2 in the call here is most likely wrong, should be nobs1

problem: all PASS examples have nobs1=nobs2
no unit tests for nobs_ratio != 1